homeassistant/helpers/entity_component.py

```diff
@@ -89,12 +89,13 @@ def __init__(
 
         self.config: ConfigType | None = None
 
+        domain_platform = self._async_init_entity_platform(domain, None)
         self._platforms: dict[
             str | tuple[str, timedelta | None, str | None], EntityPlatform
-        ] = {domain: self._async_init_entity_platform(domain, None)}
-        self.async_add_entities = self._platforms[domain].async_add_entities
-        self.add_entities = self._platforms[domain].add_entities
-
+        ] = {domain: domain_platform}
+        self.async_add_entities = domain_platform.async_add_entities
+        self.add_entities = domain_platform.add_entities
+        self._entities: dict[str, entity.Entity] = domain_platform.domain_entities
```

Comment:
This is the only functional change here, the rest is collapsing the duplicate dict lookup to make it more readable